lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -500,6 +516,14 @@ def non_tract_locations(resource: dict):
             encounter_date = datetime.strptime(record.get('time_test_order'),
                 '%Y-%m-%d %H:%M:%S').strftime('%Y-%m-%d')
 
+        # The 'Testing Determination - Internal' instrument does not capture a date,
+        # so calculate encounter_date based on the repeat instance number and the
+        # date that the first repeat instance represents for the project.
+        elif record.get('surge_selected'):
+            encounter_date = (husky_project.repeat_instance_1_date + \
+            relativedelta(days =+ (int(record.get('redcap_repeat_instance')) - 1))) \
+                .strftime('%Y-%m-%d')
+
```

Comment:
IME, datetime math used for purposes like this is sometimes prone to subtle edge cases. Instead of addressing the symptom, could we address the problem and change REDCap to capture a date? Or use one of the existing dates from the other instruments in the same repeated event instance?

Reply:
I will ask if they can add a date to the Testing Internal survey. The surge scenario could mean that the encounter event in REDCap gets created and the only thing populated is the name of a surge on the Testing Internal survey. In that case we wouldn't have another instrument to grab the date from.

Reply:
Ah, got it! Thanks for explaining. That makes sense. Hopefully a date can be added to REDCap.

Reply:
Woo hoo! They're going to add the field. I'll hold this code change. I'll take out the code I added to handle the repeated instance date once that field gets in.

Reply:
We got the date field, and I've updated the code to use that instead.